File: benchmark/programs/transfer.py
from dataclasses import dataclass

@dataclass
class Account:
    id: int
    balance: int
    owner: str

def transfer(amount: int, from_acct: Account, to_acct: Account) -> None:
    assert from_acct.balance >= amount
    from_acct.balance -= amount
    to_acct.balance += amount

File: benchmark/programs/transfer.rs
struct Account {
    id: u64,
    balance: i64,
    owner: String,
}

fn transfer(amount: u64, from: &mut Account, to: &mut Account) {
    assert!(from.balance >= amount as i64);
    from.balance -= amount as i64;
    to.balance += amount as i64;
}
